packages/back-end/types/scim.d.ts

```diff
@@ -0,0 +1,93 @@
+import { Request } from "express";
+import { ApiRequestLocals } from "./api";
+
+export type BaseScimRequest = Request & ApiRequestLocals;
+
+export interface ScimEmail {
+  primary: boolean;
+  value: string;
+  type: string;
+  display: string;
+}
+
+export interface ScimUser {
+  schemas: ["urn:ietf:params:scim:schemas:core:2.0:User"];
+  id: string;
+  displayName: string;
+  externalId?: string;
+  userName: string;
+  name: {
+    formatted: string;
+    givenName: string;
+    familyName: string;
+  };
+  active: boolean;
+  emails: ScimEmail[];
+  groups: ScimGroup[]; // TODO: figure out groups object shape and include groups
+  meta: {
+    resourceType: string;
+  };
+}
+
+export interface ScimGroup {
+  schemas: ["urn:ietf:params:scim:schemas:core:2.0:Group"];
+  id: string;
+  displayName: string;
+  members: { value: string; display: string }[];
+  meta: {
+    resourceType: "Group";
+  };
+}
+
+export interface ScimListResponse {
+  schemas: ["urn:ietf:params:scim:api:messages:2.0:ListResponse"];
+  totalResults: number;
+  Resources: ScimUser[] | ScimGroup[];
+  startIndex: number;
+  itemsPerPage: number;
+}
+
+export interface ScimError {
```

Comment:
I think we can remove this, as it doesn't look like it's being used anywhere.

Reply:
Would like to use this to type the possible responses for the APIs, but haven't gotten it to work yet